mailbox/store/src/main/java/org/apache/james/mailbox/store/AnnotationKeyValidator.java

```diff
@@ -34,27 +35,24 @@
 
     public static final String TWO_SLASH_CHARACTER = "//";
 
-    public static boolean isValid(String input) throws MailboxException {
-        if (input == null) {
-            throw new MailboxException("Cannot parse null to a key");
-        }
+    public static void validKey(String input) throws InvalidAnnotationException {
+        Preconditions.checkNotNull(input);
         if (StringUtils.isBlank(input)) {
-            throw new MailboxException("Cannot parse an empty string to a key");
+            throw new InvalidAnnotationException(input, "Cannot parse an empty key");
         }
         String key = input.trim();
         if (!key.startsWith(SLASH_CHARACTER)) {
-            throw new MailboxException("The annotation does not start with a slash " + key);
+            throw new InvalidAnnotationException(input, "The annotation does not start with a slash ");
         }
         if (key.contains(TWO_SLASH_CHARACTER)) {
-            throw new MailboxException("The annotation contains 2 consecutive '/' character " + key);
+            throw new InvalidAnnotationException(input, "The annotation contains 2 consecutive '/' character ");
         }
         if (key.endsWith(SLASH_CHARACTER)) {
-            throw new MailboxException("The annotation ends with '/' character " + key);
+            throw new InvalidAnnotationException(input, "The annotation ends with '/' character ");
         }
         if (!NAME_ANNOTATION_PATTERN.matchesAllOf(key)) {
-            throw new MailboxException("The annotation contains non-ASCII character except some special characters " + key);
+            throw new InvalidAnnotationException(input, "The annotation contains non-ASCII character except some special characters ");
```

Comment:
Which special characters ? As a user or admin, I'd like to know what's wrong without opening the code....